lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -166,6 +168,99 @@ static bool isZeroBitType(Type type) {
   return false;
 }
 
+/// Given a set of known nested types (those supported by this pass), strip off
+/// leading unpacked types.  This strips off portions of the type that are
+/// printed to the right of the name in verilog.
+static Type stripUnpackedTypes(Type type) {
+  return TypeSwitch<Type, Type>(type)
+      .Case<InOutType>([](InOutType inoutType) {
+        return stripUnpackedTypes(inoutType.getElementType());
+      })
+      .Case<UnpackedArrayType>([](UnpackedArrayType arrayType) {
+        return stripUnpackedTypes(arrayType.getElementType());
+      })
+      .Default([](Type type) { return type; });
+}
+
+/// Output the basic type that consists of packed and primitive types.  This is
+/// those to the left of the name in verilog. implicitIntType controls whether
+/// to print a base type for (logic) for inteters or whether the caller will
+/// have handled this (with logic, wire, reg, etc).  structFieldSep is a
```

Comment:
`structFieldSep` looks like premature generalization to me, since it is always ' '.  i'd drop it and constant fold that through the code to simplify it.  The pretty printed types are likely to need more than just a newline (they'll want indentation too) so I don't think a char is going to be enough to handle this.